ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Byron.hs

```diff
@@ -924,26 +1038,44 @@ instance (ByronGiven, Typeable cfg, ConfigContainsGenesis cfg)
 -- | We intentionally ignore the hash
 instance Typeable cfg => NoUnexpectedThunks (GenTx (ByronBlockOrEBB cfg)) where
   showTypeOf _ = show (typeRep (Proxy @(GenTx (ByronBlockOrEBB cfg))))
-  whnfNoUnexpectedThunks ctxt (ByronTx tx _hash) =
-      noUnexpectedThunks ctxt tx'
-    where
-      tx' :: UseIsNormalFormNamed "ATxAux" (CC.UTxO.ATxAux ByteString)
-      tx' = UseIsNormalFormNamed tx
+  whnfNoUnexpectedThunks ctxt gtx = case gtx of
+      ByronTx _hash tx ->
+        let tx' :: UseIsNormalFormNamed "ATxAux" (CC.UTxO.ATxAux ByteString)
+            tx' = UseIsNormalFormNamed tx
+        in noUnexpectedThunks ctxt tx'
+      ByronDlg _hash cert ->
+        let cert' :: UseIsNormalFormNamed "ACertificate" (CC.Delegation.ACertificate ByteString)
+            cert' = UseIsNormalFormNamed cert
+        in noUnexpectedThunks ctxt cert'
+      ByronUpdateProposal _hash prop ->
+        let prop' :: UseIsNormalFormNamed "AProposal" (CC.Update.AProposal ByteString)
+            prop' = UseIsNormalFormNamed prop
+        in noUnexpectedThunks ctxt prop'
+      ByronUpdateVote _hash vote ->
+        let vote' :: UseIsNormalFormNamed "AVote" (CC.Update.AVote ByteString)
+            vote' = UseIsNormalFormNamed vote
+        in noUnexpectedThunks ctxt vote'
```

Comment:
You can write this (all four cases) more succinctly:
```haskell
ByronUpdateVote _hash vote -> noUnexpectedThunks ctxt (UseIsNormalFormNamed @"AVote" vote)
```

Reply:
agh thanks